library/agent/Agent.ts

```diff
@@ -609,4 +612,27 @@ export class Agent {
         return this.sendHeartbeatEveryMS;
     }
   }
+
+  private registerShutdownHandler() {
+    const shutdownEvents = ["SIGTERM", "SIGINT", "SIGBREAK", "SIGHUP"];
+
+    shutdownEvents.forEach((event) => {
+      process.on(event, () => {
+        if (this.shutdownEventCalled) {
+          return;
+        }
+        this.shutdownEventCalled = true;
+        this.onShutdown();
+      });
+    });
+  }
+
+  private async onShutdown() {
+    this.logger.log("Shutting down agent...");
+    if (performance.now() - this.lastHeartbeat > 30000) {
```

Comment:
then we'll still miss some data? would make sure that a server keeps crashing and restarting, we don't send a heartbeat every time, which seems to be the case now? so rather use the start date as reference?

Reply:
Changed